proto/teleport/lib/teleterm/v1/service.proto

```diff
@@ -504,3 +506,46 @@ message GetConnectMyComputerNodeNameRequest {
 message GetConnectMyComputerNodeNameResponse {
   string name = 1;
 }
+
+message ListUnifiedResourcesRequest {
+  string cluster_uri = 1;
+  // Kinds is a list of kinds to match against a resource's kind. This can be used in a
+  // unified resource request that can include multiple types.
+  repeated string kinds = 2;
+  // Limit is the maximum amount of resources to retrieve.
+  int32 limit = 3;
+  // StartKey is used to start listing resources from a specific spot. It
+  // should be set to the previous next_key value if using pagination, or
+  // left empty.
+  string start_key = 4;
+  string query = 5;
+  string search = 6;
+  // SortBy describes which resource field and which direction to sort by.
+  SortBy sort_by = 7;
+  // the caller is able to request access to using search_as_roles
+  bool search_as_roles = 8;
+}
```

Comment:
we will need to add `pinned_only` as well here. Check [this](https://github.com/gravitational/teleport/pull/32077/files#diff-04d701ddf274076d7060ab5a9e74cc33e438c51e953f551997715bb6bfb505fdR1847-R1849)